### ISSUE TITLE
~/.xsession-errors: Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested)

### ISSUE THREAD

**Issue by jkirk** [linuxmint/cinnamon#9322]
**Facts**
```
 * Cinnamon version: 3.8.8
 * Distribution: Debian GNU/Linux 10 (buster)
 * Graphics hardware *and* driver use: Intel UHD Graphics 620 / driver: i915
 * 64 bit (x86_64)
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog: see below

% inxi -G -S   
System:    Host: executor Kernel: 5.5.0-0.bpo.2-amd64 x86_64 bits: 64 Desktop: Cinnamon 3.8.8 
           Distro: Debian GNU/Linux 10 (buster) 
Graphics:  Device-1: Intel UHD Graphics 620 driver: i915 v: kernel 
           Display: x11 server: X.Org 1.20.4 driver: modesetting unloaded: fbdev,vesa 
           resolution: 1920x1080~60Hz, 3840x2160~60Hz 
           OpenGL: renderer: Mesa DRI Intel UHD Graphics 620 (Kabylake GT2) v: 4.5 Mesa 18.3.
 ```

**Issue**

`~/.xsession-errors:` gets spammed with the following lines:
```
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested)
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
Cinnamon warning: Log level 128: posix_spawn avoided (automatic reaping requested) (fd close requested) 
```

**Steps to reproduce**

I actually can't. I quit all applications and most of the applets, restarted cinnamon, re-login, changed themes (borders, icons, controls, desktop) but could not get rid of the continuous warning messages. What did work though was creating a new user: there the `Cinnamon warning` lines were not shown.

I need help debugging this issue. Thank you!

**Comment by mtwebster** [comment]
These can be ignored.  They're a new warning we get when launching subprocesses (which hasn't changed for us).  Hopefully before the next release we can figure out how to change this to avoid the warnings, as they are a bit excessive.  I believe it's caused by the newer verison of glib in buster (and soon to be mint 20).

**Comment by leigh123linux** [comment]
@jkirk You do realize reporting it here wont fix your issue as cinnamon-3.8.8 is EOL

**Comment by claudiux** [comment]
@mtwebster This issue is still present in LinuxMint 20 beta, Cinnamon 4.6.3. Yes, it is caused by the new version of glib. Maybe a way to solve this issue: https://github.com/rpm-software-management/createrepo_c/pull/108/files ?

**Comment by mtwebster** [comment]
@claudiux I don't think completely disabling logging is the answer.

@Mintulix You may have some applet or other extension that is causing this.  It only happens when a new process is launched under cinnamon. If you figure out which one it is I'll have a look at maybe coming up with some alternative for applets to use, though, in general, I wouldn't recommend this sort of behavior in an applet in the first place.

Honestly I was expecting a new glib version to trickle down from ubuntu by now with this fixed.

**Comment by jkirk** [comment]
@mtwebster Any idea on how to find out what is causing this log messages?

As this problem is still present in the latest version of Cinnamon, doesn't it make sense to (re)open this issue again?

**Comment by jkirk** [comment]
Holy! I was looking for an answer for months and just by chance just found out by chance, that this error message seems to be caused by the applet [Bash Sensors](https://cinnamon-spices.linuxmint.com/applets/view/231). I suspect the [`Util.spawn_async`](https://github.com/linuxmint/cinnamon-spices-applets/blob/9f364757034dc531dff464d5a98f7c97306ce5dd/bash-sensors%40pkkk/files/bash-sensors%40pkkk/applet.js#L151) call. Any idea why `Util.spawn_async` is problematic?

(I will file an issue against bash-sensors too if this can be solved in the applet (even with a workaround)).

**Comment by mtwebster** [comment]
Ok I pushed a fix in muffin to avoid some specific, common warnings.  I hate that I had to do this, but I'm not sure a fix actually *is* pending upstream so we may as well address it.

**Comment by Mintulix** [comment]
Hmm, I removed all spices with Util.spawn_async and all who update every 5 seconds, rebooted - to no avail.
Thanks for the workaround!

**Comment by claudiux** [comment]
@mtwebster @jkirk @Mintulix 
I just made this PR: https://github.com/linuxmint/cinnamon-spices-extensions/pull/284. It's an Cinnamon extension that aims to solve this problem. It works well, from the moment it is launched.

**Comment by mtwebster** [comment]
Why? The issue is fixed in muffin

**Comment by claudiux** [comment]
Because you said "I hate that I had to do this".

**Comment by Rio517** [comment]
@mtwebster  can you make a new muffin release that includes the fix for this? Maybe a version 4.6.4?
